src/hoc/resource.ts

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, ComponentType } from 'react';
+import { get, isObject, isFunction } from 'lodash';
+import { connect } from 'react-redux';
+import { getFrom } from '../helpers';
+import { fetchResource } from '../actions';
+import {
+  ServiceFunction,
+  ServiceOptions,
+  ResourceOptions,
+} from '../actions/types';
+
+interface GenericProps {
+  [propName: string]: string;
+}
+
+export const withResource = (options: {
```

Comment:
It seems good but I have the feeling we have logic that somehow is similar with the other HOC already created

Reply:
Yeah. But I figured that creating a common abstraction for both would be more complicated than this